utilities/test_suite/HOST/audio/Tensor_host_audio.cpp

```diff
@@ -0,0 +1,373 @@
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include <stdio.h>
+#include <dirent.h>
+#include <string.h>
+#include <iostream>
+#include "rpp.h"
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <time.h>
+#include <omp.h>
+#include <half/half.hpp>
+#include <fstream>
+#include <iomanip>
+
+// Include this header file to use functions from libsndfile
+#include <sndfile.h>
+
+using namespace std;
+using half_float::half;
+
+typedef half Rpp16f;
+
+void remove_substring(string &str, string &pattern)
+{
+    std::string::size_type i = str.find(pattern);
+    while (i != std::string::npos)
+    {
+        str.erase(i, pattern.length());
+        i = str.find(pattern, i);
+   }
+}
+
+void verify_non_silent_region_detection(float *detectedIndex, float *detectionLength, string testCase, int bs, char audioNames[][1000])
+{
+    fstream ref_file;
+    string ref_path = get_current_dir_name();
+    string pattern = "HOST/audio/build";
+    remove_substring(ref_path, pattern);
+    ref_path = ref_path + "REFERENCE_OUTPUTS_AUDIO/";
+    int file_match = 0;
+    for (int i = 0; i < bs; i++)
+    {
+        string current_file_name = audioNames[i];
+        size_t last_index = current_file_name.find_last_of(".");
+        current_file_name = current_file_name.substr(0, last_index);  // Remove extension from file name
+        string out_file = ref_path + testCase + "/" + testCase + "_ref_" + current_file_name + ".txt";
+        ref_file.open(out_file, ios::in);
+        if(!ref_file.is_open())
+        {
+            cerr<<"Unable to open the file specified! Please check the path of the file given as input"<<endl;
+            break;
+        }
+
+        Rpp32s ref_index, ref_length;
+        Rpp32s out_index, out_length;
+        ref_file>>ref_index;
+        ref_file>>ref_length;
+        out_index = detectedIndex[i];
+        out_length = detectionLength[i];
+
+        if((out_index == ref_index) && (out_length == ref_length))
+            file_match += 1;
+        ref_file.close();
+    }
+    std::cerr<<std::endl<<"Results for Test case: "<<testCase<<std::endl;
```

Comment:
Lets stick to any one standard, to use std::cerr or std::cout for any print statements everywhere

Reply:
And spaces before/after operators